dist/services/StructuralMetadataUtils.js

```diff
@@ -156,51 +156,40 @@ function () {
     value: function buildSMUI(allItems, duration) {
       var _this2 = this;
 
-      // Regex to match hh:mm:ss, mm:ss OR seconds(as a float)/minutes(as an int)
-      var regexHHMMSS = /^([0-9]*:[0-9]*:[0-9]*.[0-9]*)$/i;
-      var regexMMSS = /^([0-9]*:[0-9]*)$/i;
-      var regexSS = /^([0-9]*.[0-9]*)$/i; // Convert file duration to seconds
+      // Regex to match hh:mm:ss.ms / mm:ss.ms
+      var regexHHMMSS = /^([0-9]*:[0-9]*(:|.)[0-9]*.[0-9]*)$/i; // Convert file duration to seconds
```

Comment:
We really don't want the `(:|.)`. It should just be `:`.

Reply:
If we replace `(:|.)` with just `:`, the tests for parsing `mm:ss` and `mm:ss.ms` fail.
![Screenshot from 2020-01-07 17-08-08](https://user-images.githubusercontent.com/1331659/71933502-4ad82980-3170-11ea-83ae-bd1b162561a1.png)

Reply:
I think that `/^([0-9]*:[0-9]*:[0-9]*.[0-9]*)$/` works well enough for HH:MM:SS.sss case but we need to handle the other possible scenarios as well (MM:SS.sss, HH:MM:SS, and MM:SS).  The SS and SS.sss cases are already handled by the `Number(time)` check. Here's the regex I propose to handle all of the cases at once: `/^(\d{1,2}:){0,2}(\d{1,2})(\.\d{0,3})?$/`
FWIW I think the `.` also needs to be escaped in the regex to avoid it matching any character.
It would probably be good to add tests for all of these cases and maybe some negative cases as well to ensure we don't match when we don't want to.